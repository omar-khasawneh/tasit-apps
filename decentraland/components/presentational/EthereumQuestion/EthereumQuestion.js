import React from "react";
import { StyleSheet, View } from "react-native";

import { responsiveHeight } from "react-native-responsive-dimensions";
import LargeText from "@presentational/LargeText";
import Button from "@presentational/Button";
import Colors from "@constants/Colors";

export default function EthereumQuestion(props) {
  return (
    <View style={styles.container}>
      <LargeText>{`Are you new to Ethereum?`}</LargeText>
      <View style={styles.buttonView}>
        <Button title="Yep" onPress={props.onSignUp} />
      </View>
      <View style={styles.buttonView}>
        <Button title="Nope" onPress={props.onSignIn} />
      </View>
    </View>
  );
}

// TODO: Migrate me to TypeScript types
EthereumQuestion.propTypes = {
  onSignUp: PropTypes.func.isRequired,
  onSignIn: PropTypes.func.isRequired,
};

const styles = StyleSheet.create({
  buttonView: {
    flexDirection: "row",
    marginTop: responsiveHeight(4),
  },
  container: {
    alignItems: "center",
    backgroundColor: Colors.backgroundColor,
    flex: 1,
    justifyContent: "center",
  },
});
