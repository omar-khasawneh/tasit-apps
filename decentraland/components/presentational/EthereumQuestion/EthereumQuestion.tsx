import React from "react";
import { StyleSheet, View } from "react-native";

import { responsiveHeight } from "react-native-responsive-dimensions";
import LargeText from "../LargeText";
import Button from "../Button";
import Colors from "../../../constants/Colors";

const styles = StyleSheet.create({
  buttonView: {
    flexDirection: "row",
    marginTop: responsiveHeight(4),
  },
  container: {
    alignItems: "center",
    backgroundColor: Colors.backgroundColor,
    flex: 1,
    justifyContent: "center",
  },
});

interface EthereumQuestionProps {
  onSignUp: (...args: any[]) => any;
  onSignIn: (...args: any[]) => any;
}

const EthereumQuestion: React.FunctionComponent<EthereumQuestionProps> = props => {
  return (
    <View style={styles.container}>
      <LargeText>{`Are you new to Ethereum?`}</LargeText>
      <View style={styles.buttonView}>
        <Button title="Yep" onPress={props.onSignUp} />
      </View>
      <View style={styles.buttonView}>
        <Button title="Nope" onPress={props.onSignIn} />
      </View>
    </View>
  );
};

export default EthereumQuestion;
