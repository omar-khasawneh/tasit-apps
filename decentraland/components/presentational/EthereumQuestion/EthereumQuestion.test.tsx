import React from "react";
import { shallow } from "enzyme";
import EthereumQuestion from "../EthereumQuestion";

describe("EthereumQuestion", () => {
  it("renders the component", () => {
    const onSignUp = () => {};
    const onSignIn = () => {};

    expect(
      shallow(<EthereumQuestion onSignUp={onSignUp} onSignIn={onSignIn} />)
    ).toMatchSnapshot();
  });
});
