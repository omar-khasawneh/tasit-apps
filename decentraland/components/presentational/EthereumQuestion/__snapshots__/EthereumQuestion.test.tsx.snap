// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EthereumQuestion renders the component 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "black",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <LargeText>
    Are you new to Ethereum?
  </LargeText>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginTop": 53.36,
      }
    }
  >
    <Button
      onPress={[Function]}
      title="Yep"
    />
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginTop": 53.36,
      }
    }
  >
    <Button
      onPress={[Function]}
      title="Nope"
    />
  </View>
</View>
`;
